artiq/firmware/libboard_misoc/flash.rs

```diff
@@ -0,0 +1,10 @@
+mod imp {
+    use csr;
+
+    pub unsafe fn reload () -> ! {
+        csr::icap::iprog_write(1);
+        loop {}
+    }
+}
+
+pub use self::imp::*;
```

Comment:
Simply put the functions at the top level.
That pattern is only used for selecting different implementations using ``#[cfg()]`` but here you are already putting the ``#[cfg()]`` in ``lib.rs``.

Reply:
Here comes a problem. The main of firmware used the previous reboot function, and every board can use that. But I'm not sure whether every board can use ICAP.

Reply:
The reboot currently is broken and should not be used. Not supporting reboot at all on those boards is better than having a broken reboot.

Reply:
Understand. Then I will disable the reboot function on those boards.